### ISSUE TITLE
Unable to get custom json serialization for enums to work

### ISSUE THREAD

**Issue by viveksunder** [tiangolo/fastapi#1986]
Hi there! We are using pydantic to define our API models and have a use-case where certain fields stored in our database (mongodb) are enums and need to be translated into an API/user friendly representation. We tried the following in order to solve this problem -

* Define a common enum base class
* Define a common pydantic model base class
* Define a `class Config` section in the above pydantic base class that will implement serialization for our enums.

```
import enum


import pydantic
from fastapi.encoders import jsonable_encoder


class BaseEnum(enum.Enum):
    """
    A base class for all enums that require custom serialization.
    """

    def get_api_repr(self):
        """
        All inheriting classes must implement `def get_api_repr`
        """
        raise NotImplementedError


class SomeStatusEnum(BaseEnum):
    ACTIVE = "ACTIVE"
    INACTIVE = "INACTIVE"

    def get_api_repr(self):
        """
        Return an API friendly value.
        """
        if self.value == self.ACTIVE.value:
            return "Active"

        return "Inactive"


class AppBaseModel(pydantic.BaseModel):
    """
    All app models should inherit from this base class
    so they can benefit from common configuration.
    """

    class Config:
        json_encoders = {
            BaseEnum: lambda v: v.get_api_repr(),
        }


class CustomModel(AppBaseModel):
    """ This does something custom ;) """

    status: SomeStatusEnum


example = CustomModel(status=SomeStatusEnum.ACTIVE)
print(jsonable_encoder(example))

```

Output - 
```
{'status': 'ACTIVE'}
```

Would expect -
```
{'status': 'Active'}
```

### Environment

* OS: Linux
* Python version: 3.8.5
* FastAPI Version: 0.61.0
* Pydantic Version: 1.6.1

PS - We only recently started using FastAPI at work and we love it!!

**Comment by viveksunder** [comment]
Also FYI - I've identified that in https://github.com/tiangolo/fastapi/blob/cdba8481c2773bfca903c439b052199ebea1eda0/fastapi/encoders.py#L60 because for enums this returns .value instead of first checking if there is a custom encoder defined for that type that this explains why the custom encoder I added in the example above doesn't get picked up. 

Does it make more sense to check if any custom encoders for the type have been defined first before applying the default for the type? If so, I'm happy to look into contributing a PR to explore that - just wanted to check if my thinking is reasonable first.
